core/src/main/java/com/exadel/aem/toolkit/bundle/customlists/datasource/ChildResourcesDatasource.java

```diff
@@ -0,0 +1,150 @@
+    private static final String PATH = "path";
+    private static final String OFFSET = "offset";
+    private static final String LIMIT = "limit";
+    private static final String REP_PREFIX = "rep:";
+    private static final String ITEM_RESOURCE_TYPE = "itemResourceType";
+    private static final String PATH_TO_JCR_CONTENT = String.format("/%s", JCR_CONTENT);
+
+    @Reference
+    private transient ExpressionResolver expressionResolver;
+
+    @Override
+    protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) {
+        ResourceResolver resolver = request.getResourceResolver();
+
+        SlingScriptHelper sling = getScriptHelper(request);
+        DataSource ds = EmptyDataSource.instance();
+
+        if (sling != null) {
+            ExpressionHelper ex = new ExpressionHelper(expressionResolver, request);
+            Config dsCfg = new Config(request.getResource().getChild(Config.DATASOURCE));
+
+            String parentPath = ex.getString(dsCfg.get(PATH, String.class));
+            Integer offset = ex.get(dsCfg.get(OFFSET, String.class), Integer.class);
+            Integer limit = ex.get(dsCfg.get(LIMIT, String.class), Integer.class);
+
+            Resource parent = parentPath != null ? resolver.getResource(parentPath) : null;
+
+            if (parent != null) {
+                String itemRT = dsCfg.get(ITEM_RESOURCE_TYPE, String.class);
+
+                Iterator<Resource> children = parent.listChildren();
+                Iterator<Resource> resources = getValidChildren(children, resolver).iterator();
+
+                ds = createDataSource(resources, offset, limit, itemRT);
+            }
+        }
+        request.setAttribute(DataSource.class.getName(), ds);
+    }
+
+    private static SlingScriptHelper getScriptHelper(ServletRequest request) {
+        SlingBindings bindings = (SlingBindings) request.getAttribute(SlingBindings.class.getName());
+        return bindings.getSling();
+    }
+
+    private DataSource createDataSource(Iterator<Resource> resources, Integer offset, Integer limit, String itemRT) {
+        return new AbstractDataSource() {
+            @SuppressWarnings("unchecked")
+            @Override
+            public Iterator<Resource> iterator() {
+                Iterator<Resource> it = new PagingIterator<>(new FilterIterator<>(resources, o -> {
+                    String name = ((Resource) o).getName();
+                    return !name.startsWith(REP_PREFIX) && !name.equals(JCR_CONTENT);
+                }), offset, limit);
+
+                return new TransformIterator(it, o -> new ResourceWrapper((Resource) o) {
+                    @Nonnull
+                    @Override
+                    public String getResourceType() {
+                        return itemRT;
+                    }
+                });
+            }
+        };
+    }
+
+    private List<Resource> getValidChildren(Iterator<Resource> parent, ResourceResolver resolver) {
```

Comment:
The first argument looks to be "children" or "parentChildren"

Reply:
Edited this method.